consensus/istanbul/core/roundstate.go

```diff
@@ -27,21 +27,31 @@ import (
 )
 
 // newRoundState creates a new roundState instance with the given view and validatorSet
-func newRoundState(view *istanbul.View, validatorSet istanbul.ValidatorSet, preprepare *istanbul.Preprepare, pendingRequest *istanbul.Request, preparedCertificate istanbul.PreparedCertificate, hasBadProposal func(hash common.Hash) bool) *roundState {
+func newRoundState(view *istanbul.View, validatorSet istanbul.ValidatorSet, preprepare *istanbul.Preprepare, pendingRequest *istanbul.Request, preparedCertificate istanbul.PreparedCertificate, parentCommits *messageSet, hasBadProposal func(hash common.Hash) bool) *roundState {
+	// in case we passed empty parent commits, initialize them
+	if parentCommits == nil {
+		parentCommits = newMessageSet(validatorSet)
```

Comment:
Should the `validatorSet` parameter be the one from the last block?  This one seems to be the current block's one.

Reply:
Yeah so there should be no need to initialize this with the current valset, removed this step